native/common/include/jpype.h

```diff
@@ -178,8 +178,14 @@ class JPResource
 #define JP_RAISE_METHOD_NOT_FOUND(msg)      { throw JPypeException(JPError::_method_not_found, NULL, msg, JP_STACKINFO()); }
 #define JP_RAISE(type, msg)                 { throw JPypeException(JPError::_python_exc, type, msg, JP_STACKINFO()); }
 
+#ifndef PyObject_HEAD
+struct _object;
+typedef _object PyObject;
+#endif
+
 // Base utility headers
 #include "jp_javaframe.h"
+#include "jp_gc.h"
```

Comment:
why do all other classes need to know about GC? Isn't it an implementation detail or do other classes call GC directly?

Reply:
I remember there was some weird order dependency problem that I was having earlier that required it to be at that spot.  Lets see if it is resolved.... nope still there.  The kitchen sink (JPContext) demands GC.   Lets just forward declare it....  okay so module and reference queue are the users.... resolved.